docs/onboarding/02-digital-twin-definition-language.md

```diff
@@ -16,6 +17,7 @@ DTDL is the schema by which models are described. Its important to understand ho
 - Models are immutable. Once uploaded, they cannot be modified. Changes are made by creating a **new version**.
 - Models are versioned. The id contains the version number after the semi-colon
 - Models can be grouped/namespaced. The id can contain path segments (delimited by colon) to group models.
+- Marker tags are simply properties using a key/value Map as a schema, but they provide a more flexible way to add information to the twins at creation time (you have to define a tag in the model, but the actual values are provided when creating each twin). And you can perform queries filtering by these tags. 

```

Comment:
I think it will be important to distinguish tags from properties. Maybe one way is to provide a sample use case where a tag would be more appropriate than a property. 

For me, having some history with mongo and c#, I relate this to having a dictionary (or hashmap) on a document this allows the properties of the document to be determined/added at runtime (as opposed to modeling each expected property at build time). I used this in a past life for a logging message properties on a queuing system (azure was barely a thing at this time). This prevented me from having to store the message props as key value pairs without modeling them at build time. However, I easily could still query and filter on those properties (similar to how we do in app insights today). 

Would the value & marker tags have a similar application where it allows addition of properties without modeling them ahead of time? If so, that may be a way to distinguish from properties. What are the trade offs with using tags vs a property? Are queries against tags more expensive than properties? If we dont know these thats ok too.

Reply:
I agree. Technically tags are properties but they provide new ways to model and query some scenarios. Do you want me to include something else in this document (I was trying to follow the same level of details as the rest of the topics)
BWT I am planning to spend some time in chapter 4 to evaluate performance considerations regarding tags...

Reply:
Maybe just a slight rewording to drive home this point. For example...

> ... but they provide a more flexible way to add information **at runtime when a twin is created. To accomplish this, a tag property of type map is defined in the model. Then, each twin can define any key/value pair within the tag property.**

Thoughts?

Reply:
Perfect! thanks